lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -168,10 +168,36 @@ spec = do
                 , expectListItemFieldEqual 0 status InLedger
                 ]
 
-        -- Verify the wallet is now delegating
-        request @ApiWallet ctx (getWalletEp w) Default Empty >>= flip verify
-            [ expectFieldEqual delegation (Delegating (p ^. #id))
-            ]
+        -- Check that the controlled stake of the pool increases.
+        --
+        -- The wallet has 100000000000 stake, the pool originally had 1, but
+        -- we have to subtract the fees for joining the stake pool. Here let's
+        -- just assume they are less than 100.
+        eventually $ do
+            request @[ApiStakePool] ctx listStakePoolsEp Default Empty >>= flip verify
+                [ expectListItemFieldSatisfy 0 (metrics . stake)
+                    (> (100000000000 + 1 - 100))
+                ]
```

Comment:
**[change required]**

Move this expectation into a separate scenario

**Why**

This is yet a different scenarios that would deserve its own test for clarity and debugging. Try to keep scenarios focused on doing one thing.

Reply:
Done. We should perhaps be mindful about that these tests take quite a lot of time though.